packages/components/src/components/NewModal/NewModal.tsx

```diff
@@ -0,0 +1,208 @@
+import { ReactNode } from 'react';
+import styled from 'styled-components';
+import { useEvent } from 'react-use';
+import { borders, Elevation, mapElevationToBackground, spacingsPx } from '@trezor/theme';
+
+import { IconButton } from '../buttons/IconButton/IconButton';
+import { Text } from '../typography/Text/Text';
+import { H3 } from '../typography/Heading/Heading';
+import { ElevationContext } from '../ElevationContext/ElevationContext';
+import { useScrollShadow } from '../../utils/useScrollShadow';
+import { NewModalButton } from './NewModalButton';
+import { NewModalContext } from './NewModalContext';
+import { ModalAlignment, NewModalBackdrop } from './NewModalBackdrop';
+import { NewModalProvider } from './NewModalProvider';
+import type { NewModalVariant } from './types';
+import { Icon, IconType } from '../Icon/Icon';
+import { mapVariantToIconBackground, mapVariantToIconBorderColor } from './utils';
+import type { UISize } from '../../config/types';
+
+const NEW_MODAL_CONTENT_ID = 'modal-content';
+const MODAL_ELEVATION = 0;
+const ICON_SIZE = 40;
+
+type NewModalSize = Extract<UISize, 'large' | 'medium' | 'small'>;
+
+const getSize = (size: NewModalSize) => {
+    switch (size) {
+        case 'small':
+            return 600;
+        case 'medium':
+            return 680;
+        case 'large':
+            return 750;
+    }
+};
+
+const Container = styled.div<{ $elevation: Elevation; $size: NewModalSize }>`
+    display: flex;
+    flex-direction: column;
+    position: relative;
+    border-radius: ${borders.radii.md};
+    transition: background 0.3s;
+    max-width: 95%;
+    min-width: 305px;
+    max-height: 80vh;
+    width: ${({ $size }) => getSize($size)}px;
+    overflow: hidden;
+    background: ${mapElevationToBackground};
+    box-shadow: ${({ theme }) => theme.boxShadowElevated};
+`;
+
+const Header = styled.header`
+    display: flex;
+    align-items: center;
+    word-break: break-word;
```

Comment:
<img width="676" alt="image" src="https://github.com/user-attachments/assets/97d061fc-4baf-4080-bcc8-459d77533231">

I believe close button should be on the top in case of long header

Reply:
nit: 

total edge case: 
<img width="648" alt="image" src="https://github.com/user-attachments/assets/9c1217fc-181c-46a0-980e-836edc10dcf2">


when title is toooooo long it's not possible to reach other content.

I suggest to restrict title to two lines and add ellipsis. But I think we don't have component for it yet. Similar one could be `TruncateWithTooltip`

This is not necessary to fix.

Reply:
btw same issue for back button 

<img width="625" alt="image" src="https://github.com/user-attachments/assets/908b6e17-933a-40b9-a929-0a737b7b2972">

Reply:
Btw thinking about adding tooltip for close and back buttons. WDYT?

Reply:
From what I can tell from the Figma designs, it should be centered?

Reply:
I see, OK. This might be question for @Hermez-cz

Reply:
I believe it won't be such an issue if I restrict the title length